float32 increment
---
bool success
